package/MDAnalysis/core/topologyobjects.py

```diff
@@ -515,6 +518,9 @@ class TopologyGroup(object):
     .. versionchanged:: 0.11.0
        Added `values` method to return the size of each object in this group
        Deprecated selectBonds method in favour of select_bonds
+    .. versionchanged:: 0.18.1
+       Empty TopologyGroup now returns correctly shaped empty array via
+       indices property and to_indices()
     """
     _allowed_types = {'bond', 'angle', 'dihedral', 'improper'}
```

Comment:
Could this be generated from the new `_BTYPE_TO_SHAPE`? This would avoid redundancy; I am sure we will end up changing one and not the other. Same goes for `self.btype` bellow.

Reply:
The chunk below defining `self.btype` automatically wouldn't be clean because dihedral and improper share 4, but I did a little clean up